okio-files/src/commonMain/kotlin/okio/Filesystem.kt

```diff
@@ -15,6 +15,74 @@
  */
 package okio
 
+/**
+ * Read and write access to a hierarchical collection of files, addressed by [paths][Path]. This
+ * is a natural interface to the [current computer's local filesystem][SYSTEM].
+ *
+ * Not Just the Local Filesystem
+ * -----------------------------
+ *
+ * Other implementations are possible:
+ *
+ *  * `FakeFilesystem` is an in-memory filesystem suitable for testing. Note that this class is
+ *    included in the `okio-testing` artifact.
+ *
+ *  * A ZIP filesystem could provide access to the contents of a `.zip` file.
+ *
+ *  * A remote filesystem could access files over the network.
+ *
+ *  * A decorating filesystem could apply monitoring, encryption, compression, or filtering to
+ *    another filesystem implementation.
+ *
+ * For improved capability and testability, consider structuring your classes to dependency inject
+ * a `Filesystem` rather than using [SYSTEM] directly.
+ *
+ * Limited API
+ * -----------
+ *
+ * This interface is limited in which filesystem features it supports. Applications that need rich
+ * filesystem features should use another API, possibly alongside this API.
+ *
+ * This class cannot create special file types like hard links, symlinks, pipes, or mounts. Reading
+ * or writing these files works as if they were regular files.
+ *
+ * It cannot read or write file access control features like the UNIX `chmod` and Windows access
+ * control lists. It does honor these controls and will fail with an [IOException] if privileges
+ * are insufficient!
+ *
+ * It cannot lock files, or query which files are locked.
+ *
+ * It cannot watch the filesystem for changes.
+ *
+ * Multiplatform
+ * -------------
+ *
+ * This class supports a matrix of Kotlin platforms (JVM, Kotlin/Native, Kotlin/JS) and operating
+ * systems (Linux, macOS, and Windows). It attempts to balance working similarly across platforms
+ * with being consistent with the local operating system.
+ *
+ * This is a blocking API which limits its applicability on concurrent Node.js services. File
+ * operations will block the event loop (and all JavaScript execution!) until they complete.
+ *
+ * It supports the path schemes of both Windows (like `C:\Users`) and UNIX (like `/home`). Note that
+ * path resolution rules differ by platform.
+ *
+ * Differences vs. Java IO APIs
+ * ----------------------------
+ *
+ * The `java.io.File` class is Java's original filesystem API. The `delete` and `renameTo` methods
+ * return false if the operation failed. The `list` method returns null if the file isn't a
+ * directory or could not be listed. This class always throws `IOExceptions` when operations don't
```

Comment:
For something like delete that might fail because it got deleted in the meantime, is it worth having some optionally includable safe operators?  Maybe extension methods .deleteSafely()? or deleteIfExists()?

Reply:
Yeah, there's a bunch of convenience features that we can build on top of these primitives. createDirectories() !